lib/safer_rails_console/patches/sandbox/auto_rollback.rb

```diff
@@ -4,21 +4,16 @@ module SaferRailsConsole
   module Patches
     module Sandbox
       module AutoRollback
-
-        def self.rollback_and_begin_new_transaction
-          connection = ::ActiveRecord::Base.connection
-          connection.rollback_db_transaction
-          connection.begin_db_transaction
-        end
-
-        def self.handle_and_reraise_exception(error)
-          if error.message.include?('PG::ReadOnlySqlTransaction')
+        def self.on(error, message:)
+          if error.message.include?(message)
             puts SaferRailsConsole::Colors.color_text( # rubocop:disable Rails/Output
               'An operation could not be completed due to read-only mode.',
               SaferRailsConsole::Colors::RED
             )
           else
-            rollback_and_begin_new_transaction
+            connection = ::ActiveRecord::Base.connection
+            connection.rollback_db_transaction
+            connection.begin_db_transaction
```

Comment:
Why did we need to inline these calls?

Reply:
We didn't in practical terms though having an entrypoint that is so low-level leaks over some abstractions.
That is the reason why I went with a single entrypoint: `AutoRollback.on(e, message: 'something')`

With that said, It does have breaking changes if `rollback_and_begin_new_transaction` is called directly - which I argue it shouldn't.

I'm happy to change it back if anyone feels strongly about it

Reply:
I don't feel that strongly, but I do this is reads a little nicer with the `rollback_and_begin_new_transaction` bit extracted. We could also make that method private if that's the concern here. But again, this is fine too, I just wanted to make sure I wasn't missing a functional change that required the inlinling.

Reply:
Changed. Check https://github.com/salsify/safer_rails_console/pull/58#discussion_r1673270538